modules/apps/frontend-data-set/frontend-data-set-sample-web/src/main/resources/META-INF/resources/partials/controlled.jsp

```diff
@@ -21,7 +21,7 @@ ControlledFDSDisplayContext controlledFDSDisplayContext = new ControlledFDSDispl
 %>
 
 <react:component
-	module="js/ControlledFrontendDataSet"
+	module="{ControlledFrontendDataSet} from frontend-data-set-sample-web"
```

Comment:
Interesting! Will we need to eventually do this for all `<react:component>`s in DXP?

Reply:
Yes. In the end everything will be rendered using ES modules and the syntax I'm using (a _de facto_ standard) to distinguish between AMD and ESM is a pseudo import statement.

So, whenever a `" from "` appears inside a module name the architecture treats it as ESM, otherwise falls back to AMD.

I'm usually more inclined to using types for this thing (ie: not using in band information to distinguish behaviours) but in this case it makes more sense to use in band rather than out of band information because:

1. We will need to support AMD and ESM in parallel for a certain amount of time (maybe months, years)
2. We want to make the transition smooth
3. When everything is finished and moved to ESM it will much easier to write a pseudo import as a String rather than creating an object to say it's ESM syntax that would be superfluous (because by that time AMD would be deprecated/removed)

Reply:
Thanks for explanations! :heart: